script.tvmaze.scrobbler/service.py

```diff
@@ -18,10 +18,18 @@
 
 from __future__ import absolute_import, unicode_literals
 
-from libs.kodi_monitor import UpdateMonitor
+import xbmc
+
+from libs.exception_logger import log_exception
+from libs.kodi_monitor import KodiMonitor
 from libs.kodi_service import logger
+from libs.scheduled_tasks import periodic_pull
 
-MONITOR = UpdateMonitor()
-logger.info('Service started')
-MONITOR.waitForAbort()
-logger.info('Service stopped')
+with log_exception():
+    xbmc.sleep(3000)
+    monitor = KodiMonitor()
+    logger.info('Service started')
+    while not monitor.abortRequested():
+        periodic_pull()
+        xbmc.sleep(60000)
```

Comment:
monitor.waitForAbort(60)

Reply:
The idea is to do a pull in 3s after Kodi start if a pull interval has passed. With your variant a pull will be executed in 60s after Kodi start. If you launch Kodi, quickly start some video and then immediately close Kodi after watching (as I often do on my Android tablet) a pull may never be executed at all (unless you enable "Pull during playback").

Reply:
`if monitor.waitForAbort(60): break`
Requires a break in the while loop...

Reply:
For some reason I was sure that `xbmc.sleep()` checks abort flag and exits prematurely if it is set. I don't know where I got that idea from. I have checked the Kodi code and this is not the case. So I'll refactor this in a different way.